3drepo/3drepo.io
# Issue 4358 - Tabular View: side panel

This fixes #4358

#### Description
In tickets table tab, the following functionalities have been added:
- Ticket editing;
- New ticket creation:
   - with default values;
   - without default values;
   - with required values that can only be set in the viewer (such as pin)
- "Open in 3d viewer" button;
- Realtime.

#### Test cases
To fully test the functionalities described above, go to the tickets table tab and:
- Click on a existing ticket and edit it in the side panel (the data should update in the respective row);
- Create a new ticket without default values (the ticket panel should originally be empty and then display in the table);
- Create a new ticket with default values that do not belong to the template (the ticket panel should originally be empty and then display in the table);
- Create a new ticket with default values that belong to the template (the ticket panel should be pre-filled and then display in the table);
- Prevent new ticket creation when the template comes with required property that can only be set in the viewer;
- Open a ticket in the viewer; (this actually depends on #4190 so you should only reach the viewer and should see `?ticketId` query param in the URL)
- Create a new ticket from the viewer in another tab and see it in the tabular view (realtime);
- Edit a ticket from the viewer in another tab and see it in the tabular view (realtime);
